code/modules/mob/mob_helpers.dm

```diff
@@ -441,6 +441,8 @@ var/list/intents = list(I_HELP,I_DISARM,I_GRAB,I_HURT)
 			C = M.original.client
 
 	if(C)
+		if(C.get_preference_value(/datum/client_preference/anon_say) == GLOB.PREF_YES)
+			return
```

Comment:
This is pretty questionable. Should we allow ghosts hide their presence with anonsay pref??

Reply:
Seems to be more or less the current behavior.